src/functional/iterator/tracing.py

```diff
@@ -211,6 +212,15 @@ def less(*args):
     return _f("less", *args)
 
 
+for math_builtin_name in builtins.MATH_BUILTINS:
+    decorator = getattr(iterator.builtins, math_builtin_name).register(TRACING)
+    exec(
+        f"""@iterator.builtins.{math_builtin_name}.register(TRACING)\n"""
+        f"""def {math_builtin_name}(*args):\n"""
+        f"""  return _f("{math_builtin_name}", *args)\n"""
+    )
```

Comment:
Is the `exec` really necessary here? Is this so that the function will have the correct value for the `__name__` attribute?

Reply:
Yes, this is the motivation. I am not particularly happy about this, but I didn't find a cleaner solution. If we refactor the iterator part we should certainly get rid of such constructs.

Reply:
I don't see the value, I propose to do the same as in embedded.

Reply:
btw, this style could be extended to almost all other builtins here